tap_appsflyer/__init__.py

```diff
@@ -147,11 +147,16 @@ def parse_source_from_url(url):
                       giveup=giveup,
                       factor=2)
 @utils.ratelimit(10, 1)
-def request(url, params=None):
+def request(url, api_token, params=None):
 
     params = params or {}
     headers = {}
 
+    if len(api_token) > 36:
+        headers["Authorization"] = "Bearer " + api_token
+    else:
+        params["api_token"] = api_token
+
```

Comment:
Since as per [AppsFlyer docs](https://support.appsflyer.com/hc/en-us/articles/12399683708305-Bulletin-API-token-changes) v2 token is a JWT token. We can validate V2 token using regex.

e.g.  `jwt_token_regex = re.compile('(^[\w-]+\.[\w-]+\.[\w-]+$)')`

Reply:
Ok. My intention there was not to validate the v2 token but to distinguish between v1 (length = 36 chars) and v2 (more than 36 chars).

Reply:
@toini is there any documentation to support this condition that `v1 is length=36 chars` and `v2 is length>36 chars`? In our internal reviews we are not sure about `api_token length` as a parameter to distinguish between token types.

Reply:
@RushiT0122 you can easily see that v1 token is a GUID (maybe you can even find documentation to confirm that). V2 token is JWT based (https://support.appsflyer.com/hc/en-us/articles/12399683708305-Bulletin-API-token-changes) so it must be longer than that (https://fusionauth.io/learn/expert-advice/tokens/jwt-components-explained).